frontend/src/app/shared/components/file-links/file-link-list/file-link-list.html

```diff
@@ -4,11 +4,11 @@
   <op-storage-information
     *ngIf="showInformationBox$ | async"
     class="op-files-tab--storage-info-box"
+    data-qa-selector="op-files-tab--storage-information"
     [infoIcon]="informationBoxIcon"
     [infoTextHeader]="informationBoxHeader"
     [infoTextContent]="informationBoxContent"
-    [buttonText]="informationBoxButton"
-    (buttonClickEventEmitter)="openStorageLocation()"
+    [buttons]="buttons | async"
```

Comment:
It feels a little bit weird to pass components via input to a child template. Why isn't the whole button logic in the `op-storage-information` component itself? From a gut feeling I'd say that it belongs to the component itself and not to the file-link-list. Further, with the current approach you can basically not re-use the component, as you'd need to duplicate the button logic.

Reply:
This is exactly the point why I introduced this model. I decoupled the logic of any button from it's implementation. I assumed for the start of this refactoring that a button has a label and some sort of action that is triggered once you click it. Therefore I created a model class that take exactly these two information. A string for the label and a function for the button action. The logic these buttons now have is passed in from the component that actually defines the use cases. In this case the `FileLinkListComponent`. This component just has a `StorageInformation` box. This component can now be completely dump and can be reused anywhere. The `FileLinkListComponent` defines in which state the label is and what the button does. Further it reduces code and complexity in the templates an keep it in the specific component.

Please contact me if I couldn't explain the concept well enough :).

Reply:
1. Yes we have a case where there is no button, when a connection error occurred to Nextcloud. Then it makes no sense to either open Nextcloud or retry authentication.
2. Yes, as this element was introduced with this feature but reside inside the tabs section, it will get reused at other places. And don"t have to be changed at all. Just define the array with the button models and done.

Further additions to the button model could be a function that can determine it the action can be executed (`canExecute:() => boolean`) and be bind to a disabled/enabled state of the button in the template. This is general a good concept to decouple logic from it's representation regarding interactable UI elements.

Reply:
okay, it's fine for me then 👍 Thanks for the explanation :)

Reply:
Very welcome.